src/datumaro/components/algorithms/hash_key_inference/hashkey_util.py

```diff
@@ -172,3 +174,30 @@ def calculate_hamming(B1, B2):
     :return: hamming distance [r]
     """
     return np.count_nonzero(B1 != B2, axis=1)
+
+
+def match_query_path(query_path, dataset):
+    query_path = os.path.expanduser(query_path)
+    common_base = os.path.commonpath([query_path, dataset.data_path])
+    result_path = None
+
+    for item in dataset:
+        query_rel = os.path.relpath(query_path, common_base)
+        item_rel = os.path.relpath(item.media.path, common_base)
+
+        for q, i in zip(query_rel.split(os.sep)[::-1], item_rel.split(os.sep)[::-1]):
+            if q == i:
+                result_path = item.media.path
+    return result_path
```

Comment:
Create `DatasetItem` from the given image path and `hash_key` as well. Then, use the `hash_key` as the query.

Reply:
I wonder if we use `hash_key` as query, then we need to update `explore_topk` function to get `hash_key` as input and we already find `hash_key` in `Datasetitem` through `explore_topk`. So in my thought, in this part, we only find the proper `DatatsetItem` whether it is in the project or not. If my understanding is wrong, please let me know.

Reply:
Is there any problem to create a new `Datasetitem` from the given image path and pass it to `explore_topk()`?
https://github.com/openvinotoolkit/datumaro/blob/f100bb848f723d13acf09947042f46623450cede/src/datumaro/components/algorithms/hash_key_inference/explorer.py#L64-L68

Reply:
I see. I will test this and update about it.